lib/client.js

```diff
@@ -53,8 +54,12 @@ function Client(config = {}) {
   if (config.version) {
     this.version = config.version;
   }
-
-  this.connectionSettings = cloneDeep(config.connection || {});
+  // if the type of connection settings is a function, pass it to the next layer
```

Comment:
and the comment `if the type of connection settings is a function, pass it to the next layer` doesn't make sense anymore either

Reply:
removed